code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -493,50 +500,112 @@
 		return 0 //under effects of time magick
 
 	..()
-
-	//Feeding, chasing food, FOOOOODDDD
-	if(!stat && !resting && !locked_to && (ckey == null))
-		turns_since_scan++
-		if(turns_since_scan > 5)
-			turns_since_scan = 0
-			if((movement_target) && !(isturf(movement_target.loc) || ishuman(movement_target.loc) ))
-				movement_target = null
-				stop_automated_movement = 0
-			if( !movement_target || !(movement_target.loc in oview(src, 3)) )
-				movement_target = null
-				stop_automated_movement = 0
-				for(var/obj/item/weapon/reagent_containers/food/snacks/S in oview(src,3))
-					if(isturf(S.loc) || ishuman(S.loc))
-						movement_target = S
+	
+	if(!stat && !resting && !locked_to && (ckey == null)) //Behavior mechanisms (om nom :3)
+		if(ian_status == IDLE)
+			get_target()
+			stop_automated_movement = 0		
+			
+		else if(ian_status == BEGIN_FOOD_HUNTING)
+			ian_status = FOOD_HUNTING
+			spawn(0) // Separate process
+				stop_automated_movement = 1
+				var/failedsteps
+				while(1)
+					if(!movement_target || src.Adjacent(movement_target) || get_dist(src, movement_target) >= 7)
+						break
+					var/stepsuccess = step_towards(src,movement_target,1)
+					if(!stepsuccess)
+						failedsteps++
+					if(failedsteps > 3)
 						break
-			if(movement_target)
-				spawn(0)
+					sleep(4)
+
+				if(movement_target)		
+					if (movement_target.loc.x < src.x)
+						dir = WEST
+					else if (movement_target.loc.x > src.x)
+						dir = EAST
+					else if (movement_target.loc.y > src.y)
+						dir = NORTH
+					else
+						dir = SOUTH
+
+					if(isturf(movement_target.loc) && src.Adjacent(movement_target))
+						movement_target.attack_animal(src)
+					else if(ishuman(movement_target.loc))
+						if(prob(20))
+							emote("me", 1, "stares at [movement_target.loc]'s [movement_target] with a sad puppy-face and whimpers.")	
+				ian_status = IDLE
+				movement_target = null
+							
+		else if(ian_status == BEGIN_POINTER_FOLLOWING)
+			ian_status = POINTER_FOLLOWING
+			if(prob(35) || pointer_caller == master)
+				spawn(0) // Separate process
 					stop_automated_movement = 1
-					step_to(src,movement_target,1)
-					sleep(3)
-					step_to(src,movement_target,1)
-					sleep(3)
-					step_to(src,movement_target,1)
-
-					if(movement_target)		//Not redundant due to sleeps, Item can be gone in 6 decisecomds
+					var/failedsteps
+					while(1)
```

Comment:
Can be more explicitly written as `while(failedsteps <= 3)` and remove that block about the `if(failedsteps > 3)`